### PR TITLE
[RTL] Wine-sync preparation

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

This PR reduces the diff between our actctx.c and wine's as a preparation to do a proper sync.

## Tests
✅ KVM x86: https://reactos.org/testman/compare.php?ids=94960,94964,94969,94971
✅ KVM x64: https://reactos.org/testman/compare.php?ids=94992,94996,94997,94998

### CODE REVIEW

File: sdk/lib/rtl/actctx.c [hunk 1]
```diff
@@ -43,6 +59,47 @@ BOOLEAN RtlpNotAllowingMultipleActivation;
...
+    OLEMISC_NOUIACTIVATE                 = 0x4000,
+    OLEMISC_ALIGNABLE                    = 0x8000,
+    OLEMISC_SIMPLEFRAME                  = 0x10000,
+    OLEMISC_SETCLIENTSITEFIRST           = 0x20000,
+    OLEMISC_IMEMODE                      = 0x40000,
+    OLEMISC_IGNOREACTIVATEWHENVISIBLE    = 0x80000,
+    OLEMISC_WANTSTOMENUMERGE             = 0x100000,
+    OLEMISC_SUPPORTSMULTILEVELUNDO       = 0x200000
+} OLEMISC;
+#endif // __REACTOS__
```
Comment (oleg-dubinskiy) [inline sdk/lib/rtl/actctx.c:101]:
```suggestion
#endif // !__REACTOS__
```

Since it corresponds to `#ifndef` (if **not** defined).

File: sdk/lib/rtl/actctx.c [hunk 2]
```diff
@@ -836,6 +884,18 @@ static inline BOOL isxmlspace( WCHAR ch )
...
+static inline const char* debugstr_xmlstr(const xmlstr_t* str)
+{
+    return debugstr_wn(str->ptr, str->len);
+}
+
+static inline const char* debugstr_version(const struct assembly_version *ver)
+{
+    return wine_dbg_sprintf("%u.%u.%u.%u", ver->major, ver->minor, ver->build, ver->revision);
+}
+#endif // __REACTOS__
```
Comment (oleg-dubinskiy) [inline sdk/lib/rtl/actctx.c:897]:
```suggestion
#endif // !__REACTOS__
```

Same here.

**Review by oleg-dubinskiy — APPROVED** [review]
In other LGTM, although these almost look like a stylish-only changes.

File: sdk/lib/rtl/actctx.c [hunk 3]
```diff
@@ -23,6 +23,21 @@
...
+#define GetCurrentProcess() NtCurrentProcess()
+#define FIXME DPRINT1
+#define WARN DPRINT1
+#define TRACE DPRINT
+#define __TRY _SEH2_TRY
+#define __EXCEPT_PAGE_FAULT _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+#define __ENDTRY _SEH2_END
+#define FILE_END_OF_FILE_INFORMATION FILE_STANDARD_INFORMATION
+#define FileEndOfFileInformation FileStandardInformation
+#define RELATIVE_PATH RtlPathTypeRelative
```
Comment (oleg-dubinskiy) [inline sdk/lib/rtl/actctx.c:36]: Are these definitions added because Wine's user mode headers are not available here? Is this part of RTL actually compiling for kernel mode only? If not, then maybe we can include an appropriate the corresponding Wine header instead? E. g., wine/debug.h, wine/exception.h and so on?

Comment (HBelusca) [reply sdk/lib/rtl/actctx.c:36]: This is because Wine doesn't understand that RTL code (in ntdll and ntoskrnl) must not use higher-level functions from Win32.

File: sdk/lib/rtl/actctx.c [hunk 4]
```diff
@@ -23,6 +23,21 @@
...
+#define GetProcessHeap() RtlGetProcessHeap()
+#define GetCurrentProcess() NtCurrentProcess()
+#define FIXME DPRINT1
+#define WARN DPRINT1
+#define TRACE DPRINT
+#define __TRY _SEH2_TRY
+#define __EXCEPT_PAGE_FAULT _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+#define __ENDTRY _SEH2_END
+#define FILE_END_OF_FILE_INFORMATION FILE_STANDARD_INFORMATION
+#define FileEndOfFileInformation FileStandardInformation
```
Comment (HBelusca) [inline sdk/lib/rtl/actctx.c:35]: hmmm why is this needed? (FileEndOfFileInformation is an info class that actually exists)

File: sdk/lib/rtl/actctx.c (hunk 3 above)
Comment (oleg-dubinskiy) [reply sdk/lib/rtl/actctx.c:36]: Ah, OK then. Thanks.

Comment (tkreuzer) [reply sdk/lib/rtl/actctx.c:36]: This is user mode only. I'll have a look at whether we can include those headers.

Comment (tkreuzer) [reply sdk/lib/rtl/actctx.c:36]: "wine/exception.h" works, "wine/debug.h" isn't that simple, as it needs quite a few additional changes and linking to libwine. This is something I would prefer to do in a different PR, because it has functional implications.

File: sdk/lib/rtl/actctx.c (hunk 4 above)
Comment (tkreuzer) [reply sdk/lib/rtl/actctx.c:35]: This is from https://github.com/reactos/reactos/commit/e2e8b52413e940619675278fe98b9a93ed0eb221
```
- Modify get_manifest_in_manifest_file function to use FileStandardInformation vice FileEndOfFileInformation (currently unsupported) in call to NtQueryInformationFile.
```
And yes, it's still unimplemented.
Looking into it.

Comment (tkreuzer) [reply sdk/lib/rtl/actctx.c:35]: Turns out this isn't implemented on Windows either, at least not in user mode.

File: sdk/lib/rtl/actctx.c [hunk 5]
```diff
@@ -22,6 +22,19 @@
 #include <debug.h>
 
 #include <wine/unicode.h>
+#include "wine/exception.h"
+
+#define GetProcessHeap() RtlGetProcessHeap()
+#define GetCurrentProcess() NtCurrentProcess()
+#define FIXME DPRINT1
```
[the remark this answers is not in the document]
Comment (oleg-dubinskiy) [reply sdk/lib/rtl/actctx.c:29]: Indeed, since it's actually a diff.

File: sdk/lib/rtl/actctx.c [hunk 6]
```diff
@@ -9,7 +9,7 @@
  *                  Jacek Caban for CodeWeavers
  *                  Alexandre Julliard
  *                  Stefan Ginsberg (stefan.ginsberg@reactos.org)
- *                  Samuel Serapión
+ *                  Samuel Serapi�n
  */
 
 /* Based on Wine 3.2-37c98396 */
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply sdk/lib/rtl/actctx.c:15]: No, there is no point in changing this now, since I'm going to sync it properly later.

File: sdk/lib/rtl/actctx.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (tkreuzer) [reply sdk/lib/rtl/actctx.c:29]: This file is full of changes everywhere and this whole section is totally hacked and that should be ovious to anyone attempting to sync this.

**Review by DarkFire01 — APPROVED** [review]

File: sdk/lib/rtl/actctx.c (hunk 1 above)
Comment (tkreuzer) [reply sdk/lib/rtl/actctx.c]: I don't think that matters, anyway, I changed it.

File: sdk/lib/rtl/actctx.c (hunk 3 above)
Comment (HBelusca) [reply sdk/lib/rtl/actctx.c]: > This is user mode only. I'll have a look at whether we can include those headers.

Well this is compiled _**in**_ ntdll, which would be better to be independent on inclusions of win32-only headers.
(I would say, an exception could be if these are hidden behind wine-specific headers like those you mentioned.)

File: sdk/lib/rtl/actctx.c (hunk 4 above)
Comment (HBelusca) [reply sdk/lib/rtl/actctx.c:34]: Ah yes indeed, and I noticed that fact too a long time ago:
https://git.reactos.org/?p=reactos.git;a=blob;f=base/services/eventlog/file.c;hb=91acf823d8d4e2b6b6e8012c98b1f178922b1547#l357

Comment (tkreuzer) [reply sdk/lib/rtl/actctx.c:34]: Because every line of hack in the file makes syncing harder. If there is only one block of our code at the top, it much less of a problem as if every 3rd line has some hack in it, like it was before, which makes syncing impossible.
